### ISSUE TITLE
PyTrilinos build: error: ‘class Domi::MDMap’ has no member named ‘getTpetraMap’

### ISSUE THREAD

**Issue by hakonhagland** [trilinos/Trilinos#4210]
I am trying to build trilinos on Ubuntu 18.10 (CMake version 3.12.1, GCC version 8.2.0)

```
cd $HOME
git clone https://github.com/trilinos/Trilinos.git
cd Trilinos
mkdir build
cd build
pyenv local 2.7.15
cmake \
  -DTPL_ENABLE_MPI=ON \
  -DMPI_BASE_DIR=/usr/lib/x86_64-linux-gnu/openmpi \
  -DTrilinos_ENABLE_ALL_PACKAGES=ON \
  -DCMAKE_INSTALL_PREFIX=/opt/trilinos \
  -D Trilinos_ENABLE_PyTrilinos:BOOL=ON \
  -D BUILD_SHARED_LIBS:BOOL=ON \
  ..
make
```
This compiled for a long time (over an hour) and when almost finished:

```
[...]

 [ 96%] Building CXX object packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Domi.dir/DomiPYTHON_wrap.cpp.o
/home/hakon/Trilinos/build/packages/PyTrilinos/src/DomiPYTHON_wrap.cpp: In function ‘Teuchos::RCP<const Tpetra::Map<int, long long int> > Domi_MDMap_getTpetraMap(Domi::MDMap*, bool)’:
/home/hakon/Trilinos/build/packages/PyTrilinos/src/DomiPYTHON_wrap.cpp:4999:27: error: ‘class Domi::MDMap’ has no member named ‘getTpetraMap’; did you mean ‘getEpetraMap’?

[... and 10 more errors ...]

make[2]: *** [packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Domi.dir/build.make:70: packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Domi.dir/DomiPYTHON_wrap.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:26278: packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Domi.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

I looked at issue #3687 and tried to disable `Tpetra` by adding `cmake` option
``` 
  -D Trilinos_ENABLE_Tpetra:BOOL=OFF 
 ```
when I rerun `cmake` with this option I get:

```
[...]

Processing enabled package: NOX (Libs)
CMake Error at cmake/tribits/core/utils/CombinedOption.cmake:130 (MESSAGE):
  Error: you can not enable the option NOX_ENABLE_THYRA_TPETRA_ADAPTERS
  unless you also enable the options NOX_ENABLE_ThyraCore,
  NOX_ENABLE_ThyraTpetraAdapters.  The current option values are:

    NOX_ENABLE_ThyraCore='ON'
    NOX_ENABLE_ThyraTpetraAdapters=''

Call Stack (most recent call first):
  packages/nox/CMakeLists.txt:125 (COMBINED_OPTION)

-- Configuring incomplete, errors occurred!
```

**Comment by mhoemmen** [comment]
@wfspotz @trilinos/pytrilinos

**Comment by wfspotz** [comment]
There is a `getTpetraMap` method in `Domi::MDMap`, but it is protected with a `HAVE_TPETRA` macro.  It looks like PyTrilinos is not setting this macro properly within the Domi python extension, although I test this and haven't had problems before.  I am trying to reproduce it now.

**Comment by wfspotz** [comment]
This should have been addressed by #3731
